### DISCUSSION TITLE
Differentiate log entry "Found 1 proof" that report a proof was found the same way when self farming or pooling

### DISCUSSION THREAD

**Discussion by myITToolBox** [Chia-Network/chia-blockchain#7616] [Ideas]
Up until pools were launched, I have been monitoring and running stats based on what gets logged in the debug.log files, my monitoring is rather simple and it consists of grepping "Found 1 proofs" which prior to pools launch 1.2 version, always (in my experience) matched the number of rewards I had received i.e. "Found 1 proofs" = 1 block win

Since pool's launch, I observed that when there is a partial match on the pool, it seems to report it the same way, which to me, seems like a false positive, or at least the way run my monitoring/stats collection right now.

My question then is: is there a way to achieve differentiation between a full match on my OG plots, which means, I am probably going to earn a reward from the solo farm, compared to what is reporting from the pool. Not sure if the log could hit or say something different indicating this match was a partial from the pool?

thanks

**Comment by MaximusChia** [comment]
"Found X proofs" works for any of both, so you mus check for other strings to diferentiate.
When you get a proof of solo block you would get something like this... sry it was set on WARNING level. 

```
2021-07-02T01:56:39.766 full_node chia.full_node.full_node: WARNING  Farmer Puzzle Hash: 71da6e7d2f2d8993223fa8afd56cb0fb296b2921a2658607f80085edea4fd0bd
2021-07-02T01:56:39.766 full_node chia.full_node.full_node: WARNING  Farmer: Pool Target: {'max_height': 0, 'puzzle_hash': '0x71da6e7d2f2d8993223fa8afd56cb0fb296b2921a2658607f80085edea4fd0bd'}
2021-07-02T01:56:39.766 full_node chia.full_node.full_node: WARNING  Farmer: Request: {'challenge_chain_sp': '0xb7e1c4e085acb67b9ead5186c2896ae2f34d05f5579f4ae8d328ea116875cb42',
 'challenge_chain_sp_signature': '0x8d844e7d087fb9704f2ee52953f518cda85e7a42745ff38b2a556fe1518493fca11f09bff90fc150e8540e90baf39d760e578612ae91912de15da8eaea5b1ce2dc764da7b0f82f47b5f128db4977926daf1baf0a028c1034cd85c40c4322e99c',
 'challenge_hash': '0x10f1519e463178e8ddbaabb8194b37d2dc856a0565918aef93359299122baaf4',
 'farmer_puzzle_hash': '0x71da6e7d2f2d8993223fa8afd56cb0fb296b2921a2658607f80085edea4fd0bd',
 'pool_signature': '0xa9df78cec83d51d167d93233bcca56c2e7c03121fac65421d650a7aeebaa61ec492e52f901893ae482b733b85111e46514128791083b3a03cc001ce87cf3798e2d0a8fe396cc4b2086a3d6ca96ea56d8abe70e8ea3b8ea6609885b614680bd2f',
 'pool_target': {'max_height': 0, 'puzzle_hash': '0x71da6e7d2f2d8993223fa8afd56cb0fb296b2921a2658607f80085edea4fd0bd'},
 'proof_of_space': {'challenge': '0x10dedad508d495876225f12115242449f1cfe25ab382dda599d53b1f90cd5980',
                    'plot_public_key': '0xb65709d838ce4c8fac1d340b08039bab30af009162a91000ee49e254788d2abd60f593ca04dcb2ca07eb39bb32ca190f',
                    'pool_contract_puzzle_hash': None,
                    'pool_public_key': '0x86dcd8465bfff78d894647ca9e49f087384b7a6a9a6963f02e0c2a8c6330ba46d850c39733e2511410bba0af0982de78',
                    'proof': '0x877fd0af1e6449a4718d827d7e938eb4b7bce45a239dc03109fed7aa575b59782d237e8bb28dfa2c5d3f9f4c662c6044b8d98fe1145d373c88cb76f6ce169b1e5526779a8398deda4cdec5f99155a23a4f60b8d53836c0285050c6c1a923735fb86007d94b91a8b654e2d32e7a5355f83a3f2cdd50f733f20800e4aa9e86aeb2a444bd60d75520fe319359e8b63ef265dcb66d99df0824764bc5275d1bca68bd32dee3db49b061c38e4cda5dcc300dbbb85e993503970ab3b143c6a74fef4c9f1fb64d78563018dd23c2d2ccd0356880998b3828c26b33f8dfa8d40b5f17c1047f22b264c2936ac0df8d5232e2a2e9be0488b626486c772180e546e2b25e026f',
                    'size': 32},
 'reward_chain_sp': '0xa34715b3f7540971944d41d7188b390ad310984154997ea42d8ec53695c02652',
 'reward_chain_sp_signature': '0x83b22e6b0b7be66b41f208d700e16bc6e507250c0f0d02e6df6f7f97827f857b5e4edb0e813c4b2aeeb2180c0e31bc8f15883f11d8b5f4bb7c03a89943b81e813802684b5ae773f89bd0b67d8839c585c896ac669a49e7bf48c7c7578ed6a4ef',
 'signage_point_index': 50}
```

**Reply by myITToolBox** [reply]
Hi,
Thanks for the reply.
I'm not sure I have seen this message in the logs whenever a block was won. I currently use INFO on log settings. In order for this message to be logged, do  I need to change the log levels to WARNING? 
And with that, not entirely certain what part of it would tell me it's a pool or self farm, guessing: the puzzle hash = none?
Doable, but would it not be cleaner if the log message at INFO level gave a hint as per what kind of proof was encountered? pool / self farm?
thanks
a
